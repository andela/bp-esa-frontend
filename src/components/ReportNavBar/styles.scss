.report-navbar-container {
  display: grid;
  grid-template-columns: 100%;
  align-items: center;

  margin: 1em 0 0 0;
  padding: 0 2em;

  background-color: rgba(204, 204, 221, 0.2);
  box-sizing: border-box;
  font-size: 0.8rem;

  &::before {
    content: 'List of Automations';
    display: none;
    color: #889;
  }
  .report-navbar {
    display: grid;
    color: #667;
    grid-template-columns: max-content max-content 1fr;
    align-items: center;
    grid-gap: 0.75em;
    padding: 0.75em;
    >p {
      display: none;
      margin: 0;
    }

    .view-button {
      height: 2em;
      width: 2em;
      cursor: pointer;
      fill: #AAB;
    }
    .view-button:active, .view-button:hover {
      fill: #667;
    }
  }

}

@media only screen and (min-width:500px) {
  .report-navbar-container {
    font-size: 1rem;
    .report-navbar {
      grid-template-columns: repeat(3, max-content) 1fr;

      >p {
        display: block;
        font-size: 1.375em;
      }
    }
  }
}

@media only screen and (min-width:700px){
  .report-navbar-container {
    grid-template-columns: max-content 1fr;
    grid-gap: 3.125em;

    &::before {
      display: block;
      font-size: 1.375em;
    }

    .report-navbar {
      justify-self: right;
      width: 100%;
      max-width: 40em;
    }
  }
}

@media only screen and (min-width:800px){
  .report-navbar-container {
    grid-gap: 4.6875em;
  }
}

@media only screen and (min-width:900px){
  .report-navbar-container {
    grid-gap: 6.35em;
  }
}
