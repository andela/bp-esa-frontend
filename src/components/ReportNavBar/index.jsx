import * as React from 'react';
import PropTypes from 'proptypes';
import './styles.scss';
import FilterComponent from '../FilterComponent';

const ReportNavBar = ({ isStats, renderView, ...props }) => (
  <React.Fragment>
    {
      isStats ? (
        <div className="left-navbar">
          <React.Fragment>
            Automations stats for Today
            <img src="/drop-down.png" alt="down" />
          </React.Fragment>
        </div>
      ) : (
        <div className="report-navbar-container">
          <div className="report-navbar">
            <p>View:</p>
            <svg version="1.1" viewBox="0.0 0.0 960.0 720.0" fill="none" stroke="none" strokeLinecap="square" strokeMiterlimit="10" xmlnsXlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" id="card-icon" className="view-button" onClick={renderView('card')}>
              <clipPath id="p.0"><path d="m0 0l960.0 0l0 720.0l-960.0 0l0 -720.0z" clipRule="nonzero" /></clipPath>
              <g clipPath="url(#p.0)">
                <path fill="#000000" fillOpacity="0.0" d="m0 0l960.0 0l0 720.0l-960.0 0z" fillRule="evenodd" />
                <path fill="#000000" d="m243.45009 -0.17608136l-242.82036 0l0 194.1241l242.82036 0z" fillRule="evenodd" />
                <path stroke="#000000" strokeWidth="1.0" strokeLinejoin="round" strokeLinecap="butt" d="m243.45009 -0.17608136l-242.82036 0l0 194.1241l242.82036 0z" fillRule="evenodd" />
                <path fill="#000000" d="m601.4111 -0.17608136l-242.82034 0l0 194.1241l242.82034 0z" fillRule="evenodd" />
                <path stroke="#000000" strokeWidth="1.0" strokeLinejoin="round" strokeLinecap="butt" d="m601.4111 -0.17608136l-242.82034 0l0 194.1241l242.82034 0z" fillRule="evenodd" />
                <path fill="#000000" d="m959.372 -0.17608136l-242.82031 0l0 194.1241l242.82031 0z" fillRule="evenodd" />
                <path stroke="#000000" strokeWidth="1.0" strokeLinejoin="round" strokeLinecap="butt" d="m959.372 -0.17608136l-242.82031 0l0 194.1241l242.82031 0z" fillRule="evenodd" />
                <path fill="#000000" d="m243.45009 262.47488l-242.82036 0l0 194.12408l242.82036 0z" fillRule="evenodd" />
                <path stroke="#000000" strokeWidth="1.0" strokeLinejoin="round" strokeLinecap="butt" d="m243.45009 262.47488l-242.82036 0l0 194.12408l242.82036 0z" fillRule="evenodd" />
                <path fill="#000000" d="m601.4111 262.47488l-242.82034 0l0 194.12408l242.82034 0z" fillRule="evenodd" />
                <path stroke="#000000" strokeWidth="1.0" strokeLinejoin="round" strokeLinecap="butt" d="m601.4111 262.47488l-242.82034 0l0 194.12408l242.82034 0z" fillRule="evenodd" />
                <path fill="#000000" d="m959.372 262.47488l-242.82031 0l0 194.12408l242.82031 0z" fillRule="evenodd" />
                <path stroke="#000000" strokeWidth="1.0" strokeLinejoin="round" strokeLinecap="butt" d="m959.372 262.47488l-242.82031 0l0 194.12408l242.82031 0z" fillRule="evenodd" />
                <path fill="#000000" d="m243.45009 525.12585l-242.82036 0l0 194.12408l242.82036 0z" fillRule="evenodd" />
                <path stroke="#000000" strokeWidth="1.0" strokeLinejoin="round" strokeLinecap="butt" d="m243.45009 525.12585l-242.82036 0l0 194.12408l242.82036 0z" fillRule="evenodd" />
                <path fill="#000000" d="m602.9698 525.1262l-242.82031 0l0 196.35175l242.82031 0z" fillRule="evenodd" />
                <path stroke="#000000" strokeWidth="1.0" strokeLinejoin="round" strokeLinecap="butt" d="m602.9698 525.1262l-242.82031 0l0 196.35175l242.82031 0z" fillRule="evenodd" />
                <path fill="#000000" d="m959.372 525.12585l-242.82031 0l0 194.12408l242.82031 0z" fillRule="evenodd" />
                <path stroke="#000000" strokeWidth="1.0" strokeLinejoin="round" strokeLinecap="butt" d="m959.372 525.12585l-242.82031 0l0 194.12408l242.82031 0z" fillRule="evenodd" />
              </g>
            </svg>
            <svg height="512pt" viewBox="0 -52 512.00001 512" width="512pt" xmlns="http://www.w3.org/2000/svg" className="view-button" onClick={renderView('list')} id="list-icon">
              <path d="m0 113.292969h113.292969v-113.292969h-113.292969zm30.003906-83.289063h53.289063v53.289063h-53.289063zm0 0" />
              <path d="m149.296875 0v113.292969h362.703125v-113.292969zm332.699219 83.292969h-302.695313v-53.289063h302.695313zm0 0" />
              <path d="m0 260.300781h113.292969v-113.292969h-113.292969zm30.003906-83.292969h53.289063v53.289063h-53.289063zm0 0" />
              <path d="m149.296875 260.300781h362.703125v-113.292969h-362.703125zm30.003906-83.292969h302.695313v53.289063h-302.695313zm0 0" />
              <path d="m0 407.308594h113.292969v-113.296875h-113.292969zm30.003906-83.292969h53.289063v53.289063h-53.289063zm0 0" />
              <path d="m149.296875 407.308594h362.703125v-113.296875h-362.703125zm30.003906-83.292969h302.695313v53.289063h-302.695313zm0 0" />
            </svg>
            <FilterComponent {...props} />
          </div>
        </div>
      )
    }
  </React.Fragment>
);

ReportNavBar.propTypes = {
  isStats: PropTypes.bool,
  renderView: PropTypes.func,

};

ReportNavBar.defaultProps = {
  isStats: false,
  renderView: null,
};

export default ReportNavBar;
