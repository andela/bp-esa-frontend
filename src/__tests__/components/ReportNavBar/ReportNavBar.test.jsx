import React from 'react';
import { shallow } from 'enzyme';
import ReportNavBar from '../../../components/ReportNavBar';



describe('ReportNavBar', () => {
  let getComponent;
  const filterMock = jest.fn();
  beforeAll(() => {
    getComponent = () => shallow(<ReportNavBar filter={filterMock} />);
  });
  it('should render as expected', () => {
    expect(getComponent()).toMatchSnapshot();
  });
});
