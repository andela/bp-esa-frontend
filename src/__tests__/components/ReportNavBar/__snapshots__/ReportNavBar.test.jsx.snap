// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportNavBar should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReportNavBar
    filter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="report-navbar"
      >
        <FilterComponent
          filter={[MockFunction]}
        />
      </div>,
      "className": "report-navbar-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FilterComponent
          filter={[MockFunction]}
        />,
        "className": "report-navbar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "filter": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="report-navbar"
        >
          <FilterComponent
            filter={[MockFunction]}
          />
        </div>,
        "className": "report-navbar-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FilterComponent
            filter={[MockFunction]}
          />,
          "className": "report-navbar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filter": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
